src/client/auth/auth.go

```diff
@@ -145,6 +165,25 @@ func IsInvalidPrincipalError(err error) bool {
 		strings.HasSuffix(err.Error(), "\"; must start with \"robot:\" or have no \":\"")
 }
 
+const noTokenErrMsg = "no authentication metadata found in context"
+
+// NoTokenError is returned by the Auth API if the caller sent a request
+// containing no auth token.
+type NoTokenError struct{}
+
+func (e NoTokenError) Error() string {
+	return noTokenErrMsg
+}
+
+// IsNoTokenError returns true if 'err' is a NoTokenError (uses string
+// comparison to work across RPC boundaries)
+func IsNoTokenError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return err.Error() == noTokenErrMsg
+}
+
```

Comment:
For static errors like this I think it's a little more standard to just do:

```
var NoTokenError = errors.New(""no authentication metadata found in context"
```

and then you don't need an `IsNoTokenError` because you can just do `err == NoTokenError`. I'm basing this off of how `io.EOF` works in the standard library.

Reply:
My main concern here is that I don't know if this'll work across RPC boundaries (which, at least for now, is part of what this library is supposed to do, and why it's in `client`).

Like one example of how this works now is that all of these `IsXYZError` methods use `strings.Contains()` instead of `strings.Equals()`. Then, if the method gets a wrapped grpc errors (where `Error()` yield `rpc error: code = Unknown desc = <actual description>`) it'll just ignore the prefix and probably do the right thing.

Reply:
I see, yeah doing with a string comparison makes sense then. You could still have it be a static error I think, rather than make a new one each time, but it doesn't really matter. I'm not sure I understand the need to put it in the client library though. Does other client code call this method?